### PR TITLE
add initial scribe CLI with examples

### PR DESCRIPTION
**Pull request by sallyom**
Signed-off-by: Sally O'Malley <somalley@redhat.com>

**Describe what this PR does**
Adds `scribe` command line interface 

**Is there anything that requires special attention?**
Will be actively developing after initial merge

**TODO in followup**
* add unit-tests
* add a list of TODOs
* add full sync command

### CODE REVIEW

File: pkg/cmd/destination.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+	"k8s.io/klog/v2"
+	kcmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+
+	scribev1alpha1 "github.com/backube/scribe/api/v1alpha1"
+)
+
+var (
+	scribeNewDestinationLong = templates.LongDesc(`
```
Comment (JohnStrunk) [inline pkg/cmd/destination.go:21]: These come out as pretty big walls of text. Could they be line wrapped like https://github.com/kubernetes/kubectl/blob/0149779a03735a5d483115ca4220a7b6c861430c/pkg/cmd/get/get.go#L88

File: cmd/scribe/scribe.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	scribeVersion = "0.0.0"
+)
+
+// this function is copied from oc/cmd/oc/oc.go
+func injectLoglevelFlag(flags *pflag.FlagSet) {
+	from := goflag.CommandLine
+	if flag := from.Lookup("v"); flag != nil {
+		level := flag.Value.(*klog.Level)
+		levelPtr := (*int32)(level)
+		flags.Int32Var(levelPtr, "loglevel", 0, "Set the level of log output (0-10)")
```
Comment (JohnStrunk) [inline cmd/scribe/scribe.go:27]: Do we need both of these?
```
      --loglevel int32                 Set the level of log output (0-10)
      --v int32                        Set the level of log output (0-10)
```

File: .golangci.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ linters:
     - gosimple
     - govet
     - ineffassign
-    - lll
+    #- lll scribe cli cobra command flag descriptions break this
```
[the remark this answers is not in the document]
Comment (sallyom) [reply .golangci.yml]: oh great, ok will update

File: cmd/scribe/scribe.go (hunk 2 above)
Comment (sallyom) [reply cmd/scribe/scribe.go:27]: probably not, will update

File: pkg/cmd/destination.go (hunk 1 above)
Comment (sallyom) [reply pkg/cmd/destination.go:21]: will do

File: README.md [hunk 4]
```diff
@@ -15,24 +15,59 @@ Card](https://goreportcard.com/badge/github.com/backube/scribe)](https://gorepor
...
+[script to setup a kind cluster](https://github.com/backube/scribe/blob/master/hack/setup-kind-cluster.sh).
+Then, to install `scribe` in a `kind cluster`, try this
+[script to install scribe in a kind cluster](https://github.com/backube/scribe/blob/master/hack/run-in-kind.sh).
+
+### Try Scribe in Kubernetes or Openshift
+
+Follow the steps in the [installation
 instructions](https://scribe-replication.readthedocs.io/en/latest/installation/index.html).
+Here are
+[useful commands to configure cluster storage classes](https://github.com/backube/scribe/blob/master/storage-setup.md).
```
[the remark this answers is not in the document]
Comment (sallyom) [reply README.md]: done

File: .golangci.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (sallyom) [reply .golangci.yml]: updated

File: README.md [hunk 5]
```diff
@@ -15,24 +15,59 @@ Card](https://goreportcard.com/badge/github.com/backube/scribe)](https://gorepor
 
 ## Getting started
 
-To try out Scribe,  follow the steps in the [installation
+### Try Scribe in Kind
```
[the remark this answers is not in the document]
Comment (sallyom) [reply README.md:18]: updated, ptal

File: pkg/cmd/options.go [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+func getOption(opt string) *string {
+	if len(opt) > 0 {
+		return &opt
+	}
+	return nil
+}
+
+func (o *ReplicationOptions) getCopyMethod(c *CommonOptions) (*CommonOptions, bool) {
+	// CopyMethod is always required
+	switch o.CopyMethod {
```
[the remark this answers is not in the document]
Comment (sallyom) [reply pkg/cmd/options.go]: updated

File: pkg/cmd/destination.go (hunk 1 above)
Comment (sallyom) [reply pkg/cmd/destination.go:22]: done

File: cmd/scribe/scribe.go (hunk 2 above)
Comment (sallyom) [reply cmd/scribe/scribe.go:27]: fixed, have `-v --version` (bc that's defaulted w/ cobra afaict) then `--loglevel` for the verbosity - fixed the double verbosity set and also the ambiguous `-v` for version & loglevel

File: pkg/cmd/options.go [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+		return c, true
+	}
+	return nil, false
+}
+
+func (o *ReplicationOptions) getCapacity(c *CommonOptions) *CommonOptions {
+	// Capacity not always required
+	switch {
+	case len(o.Capacity) > 0:
+		capacity := resource.MustParse(o.Capacity)
```
[the remark this answers is not in the document]
Comment (sallyom) [reply pkg/cmd/options.go:107]: panic. there's probably a return err alternative, this is how i've always seen it run, though. i'll take a look

File: pkg/cmd/options.go [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+	c.SSHUser = getOption(o.SSHUser)
+	c.Path = getOption(o.Path)
+	c.StorageClassName = getOption(o.StorageClassName)
+	c.VolumeSnapClassName = getOption(o.VolumeSnapshotClassName)
+	c.PVC = getOption(o.PVC)
+	c.Parameters = make(map[string]string)
+	if len(o.ProviderParameters) > 0 {
+		p := strings.Split(o.ProviderParameters, ",")
+		for _, kv := range p {
+			pair := strings.Split(kv, "/")
```
[the remark this answers is not in the document]
Comment (sallyom) [reply pkg/cmd/options.go]: done, updated

File: storage-setup.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
+# Tips for setting up storage in a K8s, OpenShift cluster
```
[the remark this answers is not in the document]
Comment (sallyom) [reply storage-setup.md]: added! (to docs/installation/index.rst)

**Comment by sallyom** [comment]
@JohnStrunk I addressed your fb. Also, I updated the examples to be within the docs, and *.rst.  I've verified the 2 examples I added are working, too. Thanks for your review!

**Review by JohnStrunk — APPROVED** [review]

**Pull request merged**